src/services/BackgroundSchedulerService/BackgroundSchedulerService.ts

```diff
@@ -74,18 +80,43 @@ const registerPeriodicTask = async (task: PeriodicTask, exposureNotificationServ
   }
 };
 
+/**
+ * @deprecated Replaced by `registerAndroidHeadlessExposureCheckPeriodicTask`
+ *             - transitioning from using BackgroundFetch to WorkManager
+ *             - this function will still be called when upgrading from v.1.1.3 or below
+ *             - `task` is passed in as a legacy parameter, but is no longer used within the function
+ */
+// @ts-ignore
 const registerAndroidHeadlessPeriodicTask = (task: PeriodicTask) => {
   if (Platform.OS !== 'android') {
     return;
   }
+  // BackgroundFetch is still used, only to register the headless task.
+  // Scheduling the periodic task itself handled by ExposureCheckScheduler
   BackgroundFetch.registerHeadlessTask(async ({taskId}) => {
     captureMessage('runAndroidHeadlessPeriodicTask', {taskId});
     try {
-      await task();
+      captureMessage('registerHeadlessTask: WorkManager');
+      // Stop and remove legacy periodic tasks that were scheduled using react-native-background-fetch
+      await BackgroundFetch.stop('react-native-background-fetch');
```

Comment:
Should this happen only for Android?

Reply:
Yes, this is for Android only. It's within a function that is only called by Android. iOS will continue to use BackgroudFetch in its entirety.